h2h/research/four-factor-model/data_cleaning.ipynb

```diff
@@ -0,0 +1,961 @@
+    "    player_info_dict[playername] = dict()\n",
+    "    #first, store the ones that don't change from season to season\n",
+    "    common_info =  playerinfo[playerinfo[\"DISPLAY_FIRST_LAST\"] == playername].reset_index()\n",
+    "    \n",
+    "    try:\n",
+    "        ht_lst = common_info.at[0,\"HEIGHT\"].split(\"-\")\n",
+    "    except AttributeError:\n",
+    "        ht_lst = [6, 6]\n",
+    "        \n",
+    "    try:\n",
+    "        pick = int(common_info.at[0,\"DRAFT_NUMBER\"])        \n",
+    "    except:\n",
+    "        pick = 61\n",
+    "        \n",
+    "    first_year = int(common_info.at[0,\"FROM_YEAR\"])     \n",
+    "    player_info_dict[playername][\"seasons_available\"] = set()\n",
+    "    for row in season_infos.itertuples():\n",
+    "        player_info_dict[playername][\"seasons_available\"].add(row.SEASON)\n",
+    "        player_info_dict[playername][row.SEASON] = dict()\n",
+    "        player_season = player_info_dict[playername][row.SEASON]\n",
+    "        \n",
+    "        player_season[\"FG3M\"] = row.FG3M\n",
+    "        player_season[\"FG3A\"] = row.FG3A\n",
+    "        player_season[\"FGM\"] = row.FGM\n",
+    "        player_season[\"FGA\"] = row.FGA\n",
+    "        player_season[\"FTM\"] = row.FTM\n",
+    "        player_season[\"FTA\"] = row.FTA\n",
+    "        player_season[\"OREB\"] = row.OREB\n",
+    "        player_season[\"DREB\"] = row.DREB\n",
+    "        player_season[\"AST\"] = row.AST\n",
+    "        player_season[\"TOV\"] = row.TOV\n",
+    "        player_season[\"BLK\"] = row.BLK\n",
+    "        player_season[\"STL\"] = row.STL\n",
+    "        player_season[\"BLKA\"] = row.BLKA\n",
+    "        player_season[\"PF\"] = row.PF\n",
+    "        player_season[\"PFD\"] = row.PFD\n",
+    "        player_season[\"AGE\"] = row.AGE\n",
+    "        player_season[\"HEIGHT\"] = int(ht_lst[0])*12 + int(ht_lst[1]) \n",
+    "        \n",
+    "        #now, we put in the pick # and the rookie tag + undrafted tag\n",
+    "        player_season[\"PICK\"] = pick\n",
+    "        player_season[\"UNDRAFTED\"] = 0 if pick != 61 else 1\n",
+    "        player_season[\"ROOKIE\"] = 1 if first_year == row.SEASON else 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 33,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n",
+      "Jamel Thomas 1999\n"
+     ]
+    }
+   ],
+   "source": [
+    "\"\"\"this will generate a dictionary that maps game ids to a lot of information about the game. From game id, it maps to\n",
+    "home and away, which maps to all the different (relevant (relevant means played 3 minutes in a game at some point in their\n",
```

Comment:
you can just replace relevant with your description of who the relevant players are